Yaegi 0.10.0 fails to build on 32bit arch

Hello back,

It look like that the new release (0.10.0) fails to build on 32bit arch.

See the [build status on Debian](https://buildd.debian.org/status/package.php?p=golang-github-traefik-yaegi).

### armel

(32bit, little endian)

See: https://buildd.debian.org/status/fetch.php?pkg=golang-github-traefik-yaegi&arch=armel&ver=0.10.0-1&stamp=1633592516&raw=0

```
--- FAIL: TestSwapFieldType (0.00s)
panic: runtime error: index out of range [1] with length 0 [recovered]
	panic: runtime error: index out of range [1] with length 0

    --- FAIL: TestInterpConsistencyBuild/issue-1065.go (0.01s)
panic: runtime error: index out of range [1] with length 0 [recovered]
	panic: runtime error: index out of range [1] with length 0
```

### armhf

(32bit, little endian)

See: https://buildd.debian.org/status/fetch.php?pkg=golang-github-traefik-yaegi&arch=armhf&ver=0.10.0-1&stamp=1633592636&raw=0

```
--- FAIL: TestSwapFieldType (0.00s)
panic: runtime error: index out of range [1] with length 0 [recovered]
	panic: runtime error: index out of range [1] with length 0

    --- FAIL: TestInterpConsistencyBuild/issue-1065.go (0.01s)
panic: runtime error: index out of range [1] with length 0 [recovered]
	panic: runtime error: index out of range [1] with length 0
```

### i386

(32bit, little endian)

See: https://buildd.debian.org/status/fetch.php?pkg=golang-github-traefik-yaegi&arch=i386&ver=0.10.0-1&stamp=1633592017&raw=0

```
--- FAIL: TestSwapFieldType (0.00s)
panic: runtime error: index out of range [1] with length 0 [recovered]
	panic: runtime error: index out of range [1] with length 0

    --- FAIL: TestInterpConsistencyBuild/issue-1065.go (0.01s)
panic: runtime error: index out of range [1] with length 0 [recovered]
	panic: runtime error: index out of range [1] with length 0
```

### mipsel

(32bit, little endian)

See: https://buildd.debian.org/status/fetch.php?pkg=golang-github-traefik-yaegi&arch=mipsel&ver=0.10.0-1&stamp=1633594645&raw=0

```
--- FAIL: TestSwapFieldType (0.00s)
panic: runtime error: index out of range [1] with length 0 [recovered]
	panic: runtime error: index out of range [1] with length 0

    --- FAIL: TestInterpConsistencyBuild/issue-1065.go (0.01s)
panic: runtime error: index out of range [1] with length 0 [recovered]
	panic: runtime error: index out of range [1] with length 0
```

---

I've opened this issue to further investigate and submit a PR.

Cheers!

--- a/internal/unsafe2/unsafe.go
+++ b/internal/unsafe2/unsafe.go
@@ -10,8 +10,17 @@ type dummy struct{}
 // DummyType represents a stand-in for a recursive type.
 var DummyType = reflect.TypeOf(dummy{})
 
+// the following type sizes must match their original definition in Go src/reflect/type.go.
+
 type rtype struct {
-	_ [48]byte
+	_ uintptr
+	_ uintptr
+	_ uint32
+	_ uint32
+	_ uintptr
+	_ uintptr
+	_ uint32
+	_ uint32
 }
 
 type emptyInterface struct {
@@ -20,14 +29,14 @@ type emptyInterface struct {
 }
 
 type structField struct {
-	_   int64
+	_   uintptr
 	typ *rtype
 	_   uintptr
 }
 
 type structType struct {
 	rtype
-	_      int64
+	_      uintptr
 	fields []structField
 }
 

